src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -357,6 +363,8 @@ static void verifyColumnNames(Connection conn, String sqlConfig, String sql,
               "Matched column \"{0}\" as \"{1}\" in query {2}",
               new Object[] { match, actual, sqlConfig });
           targets.remove(match);
+          // TODO(bmj): What happens with MySQL here.
+          typeMap.put(match, rsmd.getColumnType(i));
```

Comment:
This has a problem with case-sensitive UniqueKey.types maps. I think in UniqueKey, if the type map is case-sensitive then we just can't use this information. I was wrong that this approach solved the problem of matching the types to the unique key columns.